extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -781,22 +822,33 @@ public void onFetchPage(@Nonnull final Downloader downloader)
         if (ageRestricted) {
             final byte[] ageRestrictedBody = JsonWriter.string(prepareDesktopEmbedVideoJsonBuilder(
                     localization, contentCountry, videoId)
+                    .value(VIDEO_ID, videoId)
                     .done())
                     .getBytes(UTF_8);
-            nextResponse = getJsonPostResponse("next", ageRestrictedBody, localization);
+            nextResponse = getJsonPostResponse(NEXT, ageRestrictedBody, localization);
         } else {
-            nextResponse = getJsonPostResponse("next", body, localization);
+            final byte[] body = JsonWriter.string(prepareDesktopJsonBuilder(localization,
+                    contentCountry)
+                    .value(VIDEO_ID, videoId)
+                    .done())
+                    .getBytes(UTF_8);
+            nextResponse = getJsonPostResponse(NEXT, body, localization);
         }
 
-        if (!ageRestricted) {
+        if ((!ageRestricted && streamType == StreamType.VIDEO_STREAM)
+                || isAndroidClientFetchForced) {
             try {
                 fetchAndroidMobileJsonPlayer(contentCountry, localization, videoId);
             } catch (final Exception ignored) {
             }
         }
 
-        if (isCipherProtectedContent()) {
-            fetchDesktopJsonPlayerWithSts(contentCountry, localization, videoId);
+        if ((!ageRestricted && streamType == StreamType.LIVE_STREAM)
+                || isIosClientFetchForced) {
+            try {
+                fetchIosMobileJsonPlayer(contentCountry, localization, videoId);
+            } catch (final Exception ignored) {
+            }
```

Comment:
Maybe add some doc here why this is used for live streams, etc.
(Also maybe above)

Reply:
Everything is documented in the `forceFetchAndroidClient` and `forceFetchIosClient` methods (they have been renamed and their documentation is has been improved).